calicoctl/resourcemgr/resourcemgr.go

```diff
@@ -313,42 +344,53 @@ func (rh resourceHelper) GetTableTemplate(headings []string) (string, error) {
 	return buf.String(), nil
 }
 
+func (rh resourceHelper) GetObjectType() reflect.Type {
+	return rh.resourceType
+}
+
 // Apply is an un-typed method to apply (create or update) a resource.  This calls directly
 // through to the resource helper specific Apply method which will map the untyped call to
 // the typed interface on the client.
-func (rh resourceHelper) Apply(client *client.Client, resource unversioned.Resource) (unversioned.Resource, error) {
+func (rh resourceHelper) Apply(client client.Interface, resource ResourceObject) (ResourceObject, error) {
 	return rh.apply(client, resource)
 }
 
 // Create is an un-typed method to create a new resource.  This calls directly
 // through to the resource helper specific Create method which will map the untyped call to
 // the typed interface on the client.
-func (rh resourceHelper) Create(client *client.Client, resource unversioned.Resource) (unversioned.Resource, error) {
+func (rh resourceHelper) Create(client client.Interface, resource ResourceObject) (ResourceObject, error) {
 	return rh.create(client, resource)
 }
 
 // Update is an un-typed method to update an existing resource.  This calls directly
 // through to the resource helper specific Update method which will map the untyped call to
 // the typed interface on the client.
-func (rh resourceHelper) Update(client *client.Client, resource unversioned.Resource) (unversioned.Resource, error) {
+func (rh resourceHelper) Update(client client.Interface, resource ResourceObject) (ResourceObject, error) {
 	return rh.update(client, resource)
 }
 
 // Delete is an un-typed method to delete an existing resource.  This calls directly
 // through to the resource helper specific Delete method which will map the untyped call to
 // the typed interface on the client.
-func (rh resourceHelper) Delete(client *client.Client, resource unversioned.Resource) (unversioned.Resource, error) {
+func (rh resourceHelper) Delete(client client.Interface, resource ResourceObject) (ResourceObject, error) {
 	return rh.delete(client, resource)
 }
 
+// Get is an un-typed method to get an existing resource.  This calls directly
+// through to the resource helper specific Get method which will map the untyped call to
+// the typed interface on the client.
+func (rh resourceHelper) Get(client client.Interface, resource ResourceObject) (ResourceObject, error) {
+	return rh.get(client, resource)
```

Comment:
This is where I think you should do the logic to do a Get or List depending on whether the resource is qualified with a name/namespace or not.